src/deploy.ts

```diff
@@ -299,8 +388,14 @@ export async function deploy(
     throw error;
   }
 
-  // Build the project
-  await build({config}, new DeployBuildEffects(apiClient, deployId, effects));
+  // Upload the files
+  const uploadSpinner = clack.spinner();
+  uploadSpinner.start("");
+  await runAllWithConcurrencyLimit(buildFilePaths, async (path, i) => {
+    uploadSpinner.message(`${i + 1} / ${buildFilePaths!.length} ${path.slice(0, effects.outputColumns - 10)}`);
+    await apiClient.postDeployFile(deployId, join(config.output, path), path);
+  });
```

Comment:
Concurrency! Nice. Related #687 (though we still don’t show intermediate progress for each file). Also note that clack has a [`tasks`](https://github.com/natemoo-re/clack/blob/0cb17d3bed21f8c8569d91a5b1ece884ea3c7ee2/packages/prompts/src/index.ts#L801-L813) method for managing concurrent spinners.

Reply:
I think that `clack.tasks` can only handle serial tasks, and wouldn't know what to do with the concurrent ones we have here. It would be fun to get a better UI here at some point though.